compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/ConvertConvToIm2ColOp.cpp

```diff
@@ -60,14 +60,67 @@ static SmallVector<int64_t> getBasisFromShape(ArrayRef<int64_t> shape) {
   return basis;
 }
 
+// Collect all AffineDimExprs from an AffineExpr.
+static void collectDimExprs(ArrayRef<AffineExpr> exprs,
+                            DenseSet<AffineExpr> &out) {
+  for (auto &expr : exprs) {
+    if (auto dimExpr = dyn_cast<AffineDimExpr>(expr)) {
+      out.insert(dimExpr);
+    } else if (auto binOpExpr = dyn_cast<AffineBinaryOpExpr>(expr)) {
+      collectDimExprs({binOpExpr.getLHS(), binOpExpr.getRHS()}, out);
+    } else {
+      llvm::errs() << "Non-dimension expression found: " << expr << "\n";
```

Comment:
Don't `llvm::errs()` like this. `emitError()` or `notifyMatchFailure()` on some relevant nearby object if this recover, or `assert()` away if it isn't. This could spam people's logs. LLVM_DEBUG(), maybe

Reply:
Thanks for your suggestions! I've changed the codes to wrap this under the debug mode.